File: app/src/main/java/com/androidessence/cashcaretaker/core/CoreFragment.java
package com.androidessence.cashcaretaker.core;

import android.support.design.widget.CoordinatorLayout;
import android.support.v4.app.Fragment;
import android.view.View;

/**
 * Base Fragment class for all fragments in Cash Caretaker.
 *
 * Created by adam.mcneilly on 9/8/16.
 */
public abstract class CoreFragment extends Fragment {
    protected CoordinatorLayout root;

    /**
     * Retrieves all necessary elements for the Fragment.
     */
    protected abstract void getElements(View view);
}

File: app/src/main/java/com/androidessence/cashcaretaker/core/CoreFragment.kt
package com.androidessence.cashcaretaker.core

import android.support.design.widget.CoordinatorLayout
import android.support.v4.app.Fragment
import android.view.View

/**
 * Base Fragment class for all fragments in Cash Caretaker.

 * Created by adam.mcneilly on 9/8/16.
 */
abstract class CoreFragment : Fragment() {
    protected var root: CoordinatorLayout? = null

    /**
     * Retrieves all necessary elements for the Fragment.
     */
    protected abstract fun getElements(view: View)
}
